components/sync15/src/request.rs

```diff
@@ -251,17 +267,21 @@ where
         }
     }
 
+    // UPDATED TO NOT ALLOW UNSUPPORTED BATCHSTATE
+    // Determines if there is a batch
     #[inline]
     fn in_batch(&self) -> bool {
         match &self.batch {
-            BatchState::Unsupported | BatchState::NoBatch => false,
+            BatchState::NoBatch => false,
             _ => true,
         }
     }
 
+    // This puts an item into the batch queue
     pub fn enqueue(&mut self, record: &EncryptedBso) -> Result<bool> {
         let payload_length = record.payload.serialized_len();
 
+        // Checks on payload_lengths and bytes
```

Comment:
Be more descriptive. What is the intent of this code? What will help me understand what its intent is? 

Consider the difference between
`// Checks on payload_lengths and bytes`
and
`// Return early if the record can never be submitted with the current server limits.`.